File: apps/web/src/lib/api/website.ts
import { apiClient } from "./client"

export interface WebsitePreview {
  url: string
  title: string
  excerpt: string
  word_count: number
  author: string | null
}

export interface WebsiteImportResponse {
  id: string
  title: string
  status: string
  language_id: number
}

export async function previewWebsite(url: string): Promise<WebsitePreview> {
  return apiClient("/website/preview", {
    method: "POST",
    body: JSON.stringify({ url }),
  })
}

export async function importWebsite(params: {
  url: string
  title: string
  language_id: number
}): Promise<WebsiteImportResponse> {
  return apiClient("/website/import", {
    method: "POST",
    body: JSON.stringify(params),
  })
}

File: apps/backend/src/api/schemas/website.py
"""Pydantic schemas for website import API."""
from typing import Optional

from pydantic import BaseModel


class WebsitePreviewRequest(BaseModel):
    url: str


class WebsitePreviewResponse(BaseModel):
    url: str
    title: str
    excerpt: str
    word_count: int
    author: Optional[str] = None


class WebsiteImportRequest(BaseModel):
    url: str
    title: str
    language_id: int


class WebsiteImportResponse(BaseModel):
    id: str
    title: str
    status: str
    language_id: int
